packages/pn-personafisica-webapp/src/components/Contacts/AddSpecialContact.tsx

```diff
@@ -266,7 +273,27 @@ const AddSpecialContact = forwardRef<AddSpecialContactRef, Props>(
 
       // first check if contact already exists
       if (contactAlreadyExists(addressesData.addresses, value, sender.senderId, channelType)) {
-        setModalOpen(ModalType.EXISTING);
+        setModalOpen({ type: ModalType.EXISTING });
+        return;
+      }
+      // verify if the sender already has a contact associated
+      const oldAddress = addressesData.specialAddresses.find(
+        (addr) => addr.senderId === sender.senderId
+      );
+      if (oldAddress) {
+        setModalOpen({
+          type: ModalType.CONFIRM_LEGAL_ASSOCIATION,
+          data: {
```

Comment:
why do we need pass this data through state and not directly through modal properties?